cluster.go

```diff
@@ -149,14 +149,35 @@ type ClusterConfig struct {
 	// If not provided, a default dialer configured with ConnectTimeout will be used.
 	Dialer Dialer
 
+	// Scylla-specific cluster options
+	Scylla ScyllaClusterOpts
+
 	// internal config for testing
 	disableControlConn bool
 }
 
+// Scylla-specific cluster options
+type ScyllaClusterOpts struct {
+	// EnableSourcePortBasedLoadBalancing controls if the driver should make use
+	// of the native_shard_aware_transport_port and native_shard_aware_transport_port_ssl
+	// ports (if they are properly configured on the node).
+	//
+	// This option improves the process of establishing per-shard connections,
+	// but you need to make sure that it will work in your setup.
+	// See the "Configuration" section of README.md for more details.
+	EnableSourcePortBasedLoadBalancing bool
+}
+
 type Dialer interface {
 	DialContext(ctx context.Context, network, addr string) (net.Conn, error)
 }
 
+// DialerExt is an optional extension of the Dialer interface that allows specifying the source port.
```

Comment:
This is an internal helper type why would we want to export it and why would we want it to define here?

Please move it to scylla and do not export it.

Alternative method of implementing that would be passing the sourcePort by a context value.
In that case scylla.go should have functions to set it and read it from the context.

Reply:
I intended the DialerExt to be implemented by the user of gocql. A user might like to both use a custom dialer and the shard aware port. To use the shard aware port correctly, the dialer needs to know which source port to use when connecting.

I think we can go with the context value approach. The drawback of this solution that I see is that the driver doesn't know if the dialer knows how to use the context value, so that it may connect to the wrong shard. However, it's not a big deal because this implementation temporarily switches back to the non-shard-aware port when it connects to the wrong shard on the shard-aware port.

Reply:
I added a detailed comment https://github.com/scylladb/gocql/pull/52#issuecomment-696134248.

Reply:
Dropped the DialerExt and went with your suggestion (passing source port through context).